### ISSUE TITLE
Rides fail to load completely when server is restarted

### ISSUE THREAD

**Issue by dwightdurmon** [TheClowner/ccRides-Support#2]
**Describe the bug**
(Thank you for creating the bug report template!  This Mod is AWESOME!)

When restarting the server, rides seem to not be able to be completely loaded.  I'm not sure what the plugin is using to enumerate worlds, but I am using Multiverse if that makes a difference.  Also, for Multiverse, I have a couple of worlds aliased (e.g. KidZone is aliased as Kid World).  I'm wondering if the aliases are possibly interfering or if it is something else.  In any case, let me know if I can provide any additional information to help troubleshoot.

**Steps To Reproduce**
1. Install Mod 
2. Create ride (CAROUSEL in this case), including setting `BASE_LOCATION`, `EXIT_LOCATION`, `ALLOW_JOIN_AFTER_START: true`, `ALLOW_JOIN_BY_CART: true`
3. Enable ride and test
4. Restart server

In order to fix ride after reboot, you have to reset `BASE_LOCATION` and `EXIT_LOCATION`, then re-enable the ride.

**Expected behavior**
Rides should be saved and reloaded on server restart.

**Screenshots/Video**
If applicable, add screenshots/video to help explain your problem.
`n/a`

**Server Details:**
 - Spigot/Paper Version: `Paper version git-Paper-375 (MC: 1.15.2) (Implementing API version 1.15.2-R0.1-SNAPSHOT)`
 - ccRides Version: `ccRides v1.0.2`
 - Java Version: `System Info: Java 11 (OpenJDK 64-Bit Server VM 11.0.7+10-LTS) Host: Linux 3.10.0-1062.el7.x86_64 (amd64)`
- Plugin List (plugins): `Plugins (72): AdvancedEnchantments, Alchemist*, AnimationLib, AsyncWorldEdit, BCommons*, BetterFarming, BigDoors, BigDoorsOpener, BKCommonLib, Blacksmith, Builder*, ccRides, ChestSort, Citizens, CleanroomGenerator*, CraftBook, CustomGUI, DeluxeMenus, Denizen, DisplayShops, DivineDrop*, DoorCloser, dtlTradersPlus, DynamicLights, Essentials, EssentialsChat, EssentialsSpawn, FastCraft, FastLeafDecay, GadgetsMenu, GravityTubes, GriefPrevention, Harbor, HolographicDisplays, HolographicExtension, InfiniteScoreboard, ItemJoin, Jobs, JoinCommands*, JumpPads, LibsDisguises, LightAPI, LuckPerms, LWC, MovingDevApi, Multiverse-Core, Multiverse-Inventories, MyPoop, MythicMobs, NPC_Destinations, NPCDestinations_Animator, NPCDestinations_Farmer, OwnGarden, PlaceholderAPI, PortableCraftingInvs, ProtocolLib, Sentinel, ServerTutorialPlus*, SignLink, TCCoasters, TheNewEconomy, TimedRewards*, TradePlus, Train_Carts, UltimateTimber, VarLight, Vault, Vehicles, VoidGenerator, WarpSystem, WorldEdit, WorldGuard`

**Logs/Errors/Configs**
Please include relevant Logs and configuration files.

Here is the test ride (CAROUSEL) I set up (`/plugins/ccRides/Rides/test.yml`).

```
Generic:
  RideType: CAROUSEL
  Enabled: true
  ID: test
  CAPACITY: 10
  BASE_LOCATION:
    ==: org.bukkit.Location
    world: KidZone
    x: -4.5
    y: 66.0
    z: 1.5
    pitch: 44.549248
    yaw: 88.12024
  EXIT_LOCATION:
    ==: org.bukkit.Location
    world: KidZone
    x: 7.005282459791285
    y: 65.0
    z: 1.4213749510583293
    pitch: 22.499317
    yaw: 87.371155
  Start:
    MIN_PLAYERS: 1
    WAIT_TIME: 15
    ALLOW_JOIN_AFTER_START: true
    ALLOW_JOIN_BY_CART: true
  Price: 0
  End:
    TELEPORT_TO_EXIT: true
Carousel:
  Radius: 5.0
  Rotation:
    Ticks_Per_Full_Rotation: 300
    Num_Cycles: 2
    AccelerateLength: 0.3
  Height:
    Max_Change_±: 0.0
    Cycles_Per_Rotation: 1.0
  HorseMode: false
```

The error messages are:

```
[04:44:17] [Server thread/INFO]: [ccRides] Enabling ccRides v1.0.2
[04:44:17] [Server thread/INFO]: [ccRides] Main Configs Loaded
[04:44:17] [Server thread/ERROR]: [org.bukkit.configuration.serialization.ConfigurationSerialization] Could not call method 'public static org.bukkit.Location org.bukkit.Location.deserialize(java.util.Map)' of class org.bukkit.Location for deserialization
java.lang.IllegalArgumentException: unknown world
	at org.bukkit.Location.deserialize(Location.java:1123) ~[patched_1.15.2.jar:git-Paper-375]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.bukkit.configuration.serialization.ConfigurationSerialization.deserializeViaMethod(ConfigurationSerialization.java:85) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.serialization.ConfigurationSerialization.deserialize(ConfigurationSerialization.java:127) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.serialization.ConfigurationSerialization.deserializeObject(ConfigurationSerialization.java:207) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConstructor$ConstructCustomObject.construct(YamlConstructor.java:37) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:216) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObject(BaseConstructor.java:205) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructMapping2ndStep(BaseConstructor.java:465) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.SafeConstructor.constructMapping2ndStep(SafeConstructor.java:184) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructMapping(BaseConstructor.java:446) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.SafeConstructor$ConstructYamlMap.construct(SafeConstructor.java:521) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConstructor$ConstructCustomObject.construct(YamlConstructor.java:28) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:216) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObject(BaseConstructor.java:205) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructMapping2ndStep(BaseConstructor.java:465) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.SafeConstructor.constructMapping2ndStep(SafeConstructor.java:184) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructMapping(BaseConstructor.java:446) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.SafeConstructor$ConstructYamlMap.construct(SafeConstructor.java:521) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConstructor$ConstructCustomObject.construct(YamlConstructor.java:28) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:216) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObject(BaseConstructor.java:205) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructDocument(BaseConstructor.java:164) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:148) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:525) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.Yaml.load(Yaml.java:438) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConfiguration.loadFromString(YamlConfiguration.java:54) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.FileConfiguration.load(FileConfiguration.java:160) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.FileConfiguration.load(FileConfiguration.java:128) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConfiguration.loadConfiguration(YamlConfiguration.java:183) ~[patched_1.15.2.jar:git-Paper-375]
	at net.clownercraft.ccRides.config.ConfigHandler.<init>(ConfigHandler.java:87) ~[?:?]
	at net.clownercraft.ccRides.RidesPlugin.onEnable(RidesPlugin.java:52) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugin(CraftServer.java:472) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugins(CraftServer.java:386) ~[patched_1.15.2.jar:git-Paper-375]
	at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:488) ~[patched_1.15.2.jar:git-Paper-375]
	at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:298) ~[patched_1.15.2.jar:git-Paper-375]
	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:908) ~[patched_1.15.2.jar:git-Paper-375]
	at java.lang.Thread.run(Thread.java:834) [?:?]
[04:44:17] [Server thread/ERROR]: [org.bukkit.configuration.serialization.ConfigurationSerialization] Could not call method 'public static org.bukkit.Location org.bukkit.Location.deserialize(java.util.Map)' of class org.bukkit.Location for deserialization
java.lang.IllegalArgumentException: unknown world
	at org.bukkit.Location.deserialize(Location.java:1123) ~[patched_1.15.2.jar:git-Paper-375]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.bukkit.configuration.serialization.ConfigurationSerialization.deserializeViaMethod(ConfigurationSerialization.java:85) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.serialization.ConfigurationSerialization.deserialize(ConfigurationSerialization.java:127) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.serialization.ConfigurationSerialization.deserializeObject(ConfigurationSerialization.java:207) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConstructor$ConstructCustomObject.construct(YamlConstructor.java:37) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:216) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObject(BaseConstructor.java:205) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructMapping2ndStep(BaseConstructor.java:465) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.SafeConstructor.constructMapping2ndStep(SafeConstructor.java:184) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructMapping(BaseConstructor.java:446) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.SafeConstructor$ConstructYamlMap.construct(SafeConstructor.java:521) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConstructor$ConstructCustomObject.construct(YamlConstructor.java:28) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:216) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObject(BaseConstructor.java:205) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructMapping2ndStep(BaseConstructor.java:465) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.SafeConstructor.constructMapping2ndStep(SafeConstructor.java:184) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructMapping(BaseConstructor.java:446) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.SafeConstructor$ConstructYamlMap.construct(SafeConstructor.java:521) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConstructor$ConstructCustomObject.construct(YamlConstructor.java:28) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObjectNoCheck(BaseConstructor.java:216) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructObject(BaseConstructor.java:205) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.constructDocument(BaseConstructor.java:164) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:148) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:525) ~[patched_1.15.2.jar:git-Paper-375]
	at org.yaml.snakeyaml.Yaml.load(Yaml.java:438) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConfiguration.loadFromString(YamlConfiguration.java:54) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.FileConfiguration.load(FileConfiguration.java:160) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.FileConfiguration.load(FileConfiguration.java:128) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.configuration.file.YamlConfiguration.loadConfiguration(YamlConfiguration.java:183) ~[patched_1.15.2.jar:git-Paper-375]
	at net.clownercraft.ccRides.config.ConfigHandler.<init>(ConfigHandler.java:87) ~[?:?]
	at net.clownercraft.ccRides.RidesPlugin.onEnable(RidesPlugin.java:52) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugin(CraftServer.java:472) ~[patched_1.15.2.jar:git-Paper-375]
	at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugins(CraftServer.java:386) ~[patched_1.15.2.jar:git-Paper-375]
	at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:488) ~[patched_1.15.2.jar:git-Paper-375]
	at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:298) ~[patched_1.15.2.jar:git-Paper-375]
	at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:908) ~[patched_1.15.2.jar:git-Paper-375]
	at java.lang.Thread.run(Thread.java:834) [?:?]
[04:44:17] [Server thread/INFO]: [ccRides] "test" ride config loaded.
[04:44:17] [Server thread/INFO]: [ccRides] Linked into Vault for Economy
[04:44:17] [Server thread/INFO]: [ccRides] Started test Ride!
[04:44:17] [Server thread/INFO]: [ccRides] Linked PlaceholderAPI
```

**Comment by TheClowner** [comment]
This should be fixed in version 1.0.4, could you please test and confirm the issue is solved?

**Comment by dwightdurmon** [comment]
That fixed the issue!  Thank you so much for your help and the awesome plugin!
